sdk/eventhub/Azure.Messaging.EventHubs.Processor/tests/Processor/EventProcessorClientLiveTests.cs

```diff
@@ -139,241 +132,144 @@ await using (var consumer = new EventHubConsumerClient(EventHubConsumerClient.De
         /// </summary>
         ///
         [Test]
-        [Ignore("Unstable test. (Tracked by: #7458). See also, line 784.")]
         public async Task EventProcessorCanReceiveFromSpecifiedInitialEventPosition()
         {
-            await using (EventHubScope scope = await EventHubScope.CreateAsync(2))
-            {
-                var connectionString = TestEnvironment.BuildConnectionStringForEventHub(scope.EventHubName);
+            await using EventHubScope scope = await EventHubScope.CreateAsync(2);
+            var connectionString = TestEnvironment.BuildConnectionStringForEventHub(scope.EventHubName);
 
-                await using (var connection = new EventHubConnection(connectionString))
-                {
-                    int receivedEventsCount = 0;
+            await using var connection = new EventHubConnection(connectionString);
+            int receivedEventsCount = 0;
 
-                    // Send some events.
+            // Send some events.
 
-                    var expectedEventsCount = 20;
-                    DateTimeOffset enqueuedTime;
+            var expectedEventsCount = 20;
+            var firstBatchEventCount = 30;
+            DateTimeOffset enqueuedTime = DateTimeOffset.MinValue;
 
-                    await using (var producer = new EventHubProducerClient(connection))
-                    {
-                        // Send a few dummy events.  We are not expecting to receive these.
+            await using var producer = new EventHubProducerClient(connection);
+            // Send a few dummy events.  We are not expecting to receive these.
 
-                        using (var dummyBatch = await producer.CreateBatchAsync())
-                        {
-                            for (int i = 0; i < 30; i++)
-                            {
-                                dummyBatch.TryAdd(new EventData(Encoding.UTF8.GetBytes("I'm dummy.")));
-                            }
+            using (var dummyBatch = await producer.CreateBatchAsync())
+            {
+                for (int i = 0; i < firstBatchEventCount; i++)
+                {
+                    dummyBatch.TryAdd(new EventData(Encoding.UTF8.GetBytes(firstBatchBody)));
+                }
 
-                            await producer.SendAsync(dummyBatch);
-                        }
+                await producer.SendAsync(dummyBatch);
+            }
 
-                        // Wait a reasonable amount of time so the events are able to reach the service.
+            // Wait a reasonable amount of time so the events are able to reach the service.
 
-                        await Task.Delay(1000);
+            await Task.Delay(1000);
```

Comment:
Since there's an element of non-determinism here, should we turn this into a polling loop that checks to see if there is an event available, either moving on at that point or timing out after 30 seconds or so?

Reply:
I debated removing this entirely, but the reason I left it is admittedly not consistent with the comment anymore. 

Since the approach to identify the `enqueuedTime` we''ll use later on to validate processing `DefaultStartingPosition = EventPosition.FromEnqueuedTime(enqueuedTime)` is now deterministic, this delay only serves to create a time gap between the first and second batch of events. Do you think checking for an available event is a better way to ensure a time gap?

Reply:
Ah... I see.   If you're just making a gap, then I think we're good.   We've had a bunch of timing issues in the past around "wait XXX after publishing and then assume we have events to read" that we've been slowly rooting out.   

Definitely worth a comment to mention that you're just forcing a time gap.   May also be worth doing at 1.5 > 2 seconds, though we tend to have CI get slower rather than faster, so... may not be an issue.   :smile: